styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -430,8 +425,9 @@ private synchronized void tryPollPods() {
       final Optional<RunState> runState = lookupPodRunState(pod, workflowInstance.get());
       if (runState.isPresent()) {
         emitPodEvents(Watcher.Action.MODIFIED, pod, runState.get());
-      } else {
         cleanup(workflowInstance.get(), pod.getMetadata().getName());
+      } else {
+        deletePod(workflowInstance.get(), pod.getMetadata().getName());
```

Comment:
I guess a reason we didn't do remove pods with no matching WF runstate before is that (at the time) the pod execution id might not be  present in the WF runstate. Pod events / polling could thus race with the execution id persistence and incorrectly cause recently created pods to be deleted. This changed in https://github.com/spotify/styx/pull/198 and the execution id is now created before the actual pod. Let's hope this holds true in the future as well =)